File: source/domain/use_cases/providers/get_articles_by_subject_use_case.py
from abc import ABC, abstractclassmethod

class GetArticlesBySubjectData(ABC):
    subject: str
    depth: int

class GetArticlesBySubjectUseCase(ABC):
    @abstractclassmethod
    def get_by_subject(self, data: GetArticlesBySubjectData) -> list:
        pass

File: source/domain/use-cases/article/get-articles-by-subject-use-case.ts
import { Article } from '@/domain/entities'

export interface GetArticlesBySubjectData {
  subject: string
  depth?: number
  range?: number
}

export interface GetArticlesBySubjectUseCase {
  getArticlesBySubject(data: GetArticlesBySubjectData): Promise<Article[]>
}
